.github/workflows/tests.yml

```diff
@@ -2,7 +2,28 @@ name: Tests
 on: pull_request
 
 jobs:
+  linters:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry config virtualenvs.create false
+          poetry install --no-root
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files
+
   integration-tests:
+    needs: linters
```

Comment:
Let's keep the jobs running in parallel.

```suggestion
```

Reply:
Done! c3c4ae8

I'm not sure I 100% understand this approach, the job to run the linters takes ~30 seconds whereas the job to run the tests takes 5 minutes. If a commit does not conform to the linting conventions, then I don't see the advantage of the tests having started already, from a computational usage standpoint this isn't efficient. And when the tests take 5 minutes, waiting an extra 30 seconds for them to begin doesn't feel like a big ask.

Reply:
Thanks for reverting this. The main point is to always get all the reports so that you don't have to fix them one-by-one and always rerun the pipeline. This is especially helpful when you forget to fix the linters but you want to get the test results.